### DISCUSSION TITLE
Default orm mapping config for bundles

### DISCUSSION THREAD

**Discussion by chapterjason** [symfony/symfony#44201] [Q&A]
Hey :)

The PR #43181 adds a `$mappingConfig` if a folder under `config/doctrine` is found under a bundle.
In my case there are files like: `User.orm.xml` under `config/doctrine`.

But I already added this folder with `DoctrineOrmMappingsPass::createXmlMappingDriver` with another prefix in the `Bundle::build` method in my bundle.

Now the path is added twice, the default config have a "wrong" prefix which leads to a "wrong" namespace and entity files which doesn't exist.

- I added: `['[...]/config/doctrine' => 'Acme\Component\User\Model']` <- exist
- the default mapping add: `['[...]config/doctrine' => 'Acme\Bundle\User\Entity']` <- doesn't exist

(debugging this was pretty hard until I realized it was registered more than once and my pipeline was surprisingly broken without any change on my side.)

Could the default mapping be skipped or disabled in any way if the folder was already added?

Using another folder works as a workaround.

Might be that @mbabker could help me here?

**Comment by mbabker** [comment]
The quickest answer I can come up with is that if you have something that shouldn't be picked up by auto-mapping at all (which sounds like is the case here since your mapping isn't resolving to an entity within the bundle where that path is picked up), you'd need to move those files out of that path; which, is an unfortunate side effect of that feature PR, but to my knowledge, the Doctrine mapping files were the only remaining bundle resource which has some kind of core Symfony integration that didn't support the newer alternative bundle structure.

If the resources in that path should be picked up by auto-mapping, I'd need to see a bigger code example to work out what's the best path forward for your use case (be it a fix in Symfony or in your code, hard to say with certainty what the right answer is just yet).

**Reply by chapterjason** [reply]
@mbabker I've created a small reproducer with the smallest amount of files which shows my structure and issue: https://github.com/chapterjason/reproducer-orm-mapping

**Reply by mbabker** [reply]
On a really quick read, I think my initial hunch of "don't use `config/doctrine`" is the answer here.  Which, is a very unfortunate side effect of my PR.  I'm not seeing a good way to get auto-mapping support for the newer bundle structure to work while also not breaking anything that might have previously used that path with an explicit configuration that goes against the auto-mapping expectations (i.e. being conditionally enabled with a container parameter, using a namespace that's outside the bundle namespace, etc.).

**Reply by chapterjason** [reply]
@mbabker Thank you for your time. I renamed the config folder to `config/bridge/user/doctrine` which works for me.
